docs/software-architecture-and-design.md

```diff
@@ -308,64 +311,147 @@ A finite state machine (FSM) is a system with a finite number of states with def
-```
-/* omitted rows of transition table */
-SubFSM = next_state, // Transitions to next_state only when the SubFSM is in the terminal state, X
-/* omitted rows of transition table */
-```
-In order to update a subFSM with an event, we need to do the following:
+The library also supports hierarchical FSMs. Sub-FSMs are treated as states where an unconditional transition occurs when the sub-FSM is in the terminal state, `X`.
+```cpp
+const auto SubFSM_S = boost::sml::state<SubFSM>;
+
+// ...in the transition table...
+SubFSM_S = next_state // Transitions to next_state only when the SubFSM is in the terminal state, X
 ```
+In order to update a sub-FSM with an event, we need to do the following:
+```cpp
 const auto update_sub_fsm_action =
-    [](auto event, back::process<TypeOfSubFSMEvent> processEvent) {
-        TypeOfSubFSMEvent sub_fsm_event = // initialize the subFSM event
+    [](auto event, boost::sml::back::process<TypeOfSubFSMEvent> processEvent) {
+        TypeOfSubFSMEvent sub_fsm_event = // initialize the sub-FSM event
         processEvent(sub_fsm_event);
     };
+
+// ...in the transition table...
+SubFSM_S + event / update_sub_fsm_action
+// When the parent FSM is updated with an event and it is
+// in the SubFSM_S state, the update_sub_fsm_action will be
+// invoked in the parent FSM. In turn, the update_sub_fsm_action 
+// will update the SubFSM with an event by calling processEvent
 ```
 The convenience of this syntax comes at the cost of hard to read error messages due to the functor and templating system.
 
 ## How Do We Use SML?
-We use SML to manage our [Tactics](#tactic). Each state represents a stage in the tactic where the robot should be doing a particular action or looking for certain conditions to be true. An example of this is the MoveFSM. While the robot is not at the destination and oriented correctly, the FSM is in the move state. Once the robot reaches its destination, it enters the terminal state, _X_, to indicate that it's done. SML also allows us to easily reuse FSMs in other tactics. For example, if a shadowing tactic needs to move to a particular destination with a certain orientation, then it can use the MoveFSM as a sub-FSM state.
+We use SML to implement our [Plays](#plays) and [Tactics](#tactic). Each state represents a stage in the play/tactic where the team/robot should be doing a particular thing or looking for certain conditions to be true. An example of this is the `MoveFSM`. While the robot is not at the destination and oriented correctly, the FSM is in the move state. Once the robot reaches its destination, it enters the terminal state, `X`, to indicate that it's done. SML also allows us to easily reuse FSMs in other tactics. For example, if a shadowing tactic needs to move to a particular destination with a certain orientation, then it can use the MoveFSM as a sub-FSM state.
```

Comment:
```suggestion
We use SML to implement our [Plays](#plays) and [Tactics](#tactics). Each state represents a stage in the play/tactic where the team/robot should be doing a particular thing or looking for certain conditions to be true. An example of this is the `MoveFSM`. While the robot is not at the destination and oriented correctly, the FSM is in the move state. Once the robot reaches its destination, it enters the terminal state, `X`, to indicate that it's done. SML also allows us to easily reuse FSMs in other tactics. For example, if a shadowing tactic needs to move to a particular destination with a certain orientation, then it can use the MoveFSM as a sub-FSM state.
```